daemon/fqdn.go

```diff
@@ -73,11 +76,88 @@ func (d *Daemon) bootstrapFQDN(restoredEndpoints *endpointRestoreState, preCache
 		OverLimit:      option.Config.ToFQDNsMaxIPsPerHost,
 		Cache:          fqdn.NewDNSCache(option.Config.ToFQDNsMinTTL),
 		LookupDNSNames: fqdn.DNSLookupDefaultResolver,
-		AddGeneratedRules: func(generatedRules []*policyApi.Rule) error {
+		AddGeneratedRulesAndUpdateSelectors: func(generatedRules []*policyApi.Rule, selectorWithIPsToUpdate map[string][]net.IP, selectorsWithoutIPs []string) error {
```

Comment:
I'm not sure what your longer-term plans are for this, but please consider splitting this out into smaller functions. This is not unit-testable, and it's is so long that it's easy to forget you're still just initializing a struct by the end of it. (The existing code also suffers from this..)

Reply:
+1

Reply:
I must say, please write this function somewhere else instead directly here.